### PR TITLE
feat: Add gRPC metrics export support

### PR DESCRIPTION
**Pull request by cartermp**
For funsies, fixes https://github.com/honeycombio/honeycomb-opentelemetry-node/issues/128

### CODE REVIEW

File: src/exporter-utils.ts [hunk 1]
```diff
@@ -44,9 +46,14 @@ export function getHoneycombMetricReader(
     // only enable metrics if a metrics dataset has been set
     return undefined;
   }
+
+  const exporter =
```
[the remark this answers is not in the document]
Comment (cartermp) [reply src/exporter-utils.ts]: doneski

**Comment by cartermp** [comment]
All diddly-doneski

File: src/exporter-utils.ts [hunk 2]
```diff
@@ -26,12 +30,30 @@ export const OTLP_PROTO_VERSION = '0.16.0';
...
     return configureHoneycombGrpcTraceExporter(options);
   }
   return configureHoneycombHttpProtoTraceExporter(options);
 }
 
+/**
+ * Configures and returns a metric exporter based on the OTLP protocol
+ * provided via options.
+ *
+ * Defaults to a http/profobuf exporter if not configured.
```
Comment (JamieDanielson) [inline src/exporter-utils.ts:43]:
```suggestion
 * Defaults to a http/protobuf exporter if not configured.
```

File: src/grpc-metric-exporter.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  DATASET_HEADER_KEY,
+  OTLP_HEADER_KEY,
+  OTLP_PROTO_VERSION,
+  TEAM_HEADER_KEY,
+} from './exporter-utils';
+import {
+  computeOptions,
+  HoneycombOptions,
+  isClassic,
+} from './honeycomb-options';
```
Comment (JamieDanielson) [inline src/grpc-metric-exporter.ts:12]:
```suggestion
import { computeOptions, HoneycombOptions } from './honeycomb-options';
```

`isClassic` isn't needed in this one, prettier then smushes it to one line

**Review by JamieDanielson — COMMENTED** [review]
Hm for some reason the smoke test fails on metrics locally...

**Review by JamieDanielson — COMMENTED** [review]
OK, so for smoke tests, in `smoke-sdk-grpc.bats` and `smoke-sdk-grpc-ts.bats` we need to add `wait_for_metrics 15` after `wait_for_traces` in the `setup_file()`.

~But when I do that, I'm noticing an error 
{"stack":"Error: PeriodicExportingMetricReader: metrics export failed (error Error: Parse Error: Expected HTTP/)\n    at PeriodicExportingMetricReader._runOnce (/usr/src/app/node_modules/@opentelemetry/sdk-metrics/build/src/export/PeriodicExportingMetricReader.js:56:19)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Timeout._onTimeout (/usr/src/app/node_modules/@opentelemetry/sdk-metrics/build/src/export/PeriodicExportingMetricReader.js:63:17)","message":"PeriodicExportingMetricReader: metrics export failed (error Error: Parse Error: Expected HTTP/)","name":"Error"}~
eh, ignore this error, I cleared some shit locally and it works. So just the wait_for_metrics should be sufficient!

**Review by JamieDanielson — APPROVED** [review]
star trek nodding

**Pull request merged**